src/cookbook/gradual-compositionapi-migration.md

```diff
@@ -0,0 +1,258 @@
+   ></iframe>
+
+Note that the template stays the same, but we've extracted the logic to use in our script section differently. If, from here, we wanted to use `isShowing` in the Options API, we could access it with `this.isShowing` now, just as we normally do with a data property, and similarly, we can access `this.toggleShow` like we would a method.
+
+You may notice we used `reactive` here instead of `refs`. This is intentional- if you're refactoring a codebase, potentially full of many many values, `reactive` translates faster and more directly as the API is closer to Options, you're still using that same object notation.
+
+## In Place of Vuex
+
+It is possible to use the Composition API in place of Vuex, and save yourself a dependency. That said, it's not exactly necessary, and like many approaches, there are tradeoffs.
+
+If you're using Vuex, it's very clear exactly what centralized state is being used across the application. In addition, you gain the benefit of using a standardized pattern that comes with officially supported tooling in the DevTools which can make a big difference when it comes to debugging code.
+
+Composition API is very flexible, but you may lose that implicit declaration in communication to fellow maintainers. Our suggestion that if you do use it as a centralized state management store, that you place it in a `stateManagement` folder, or something similarly named, so that responsibilities are clear.
+
+## Components
+
+It is possible at this point to start to move your components to the Composition API, however, for larger code bases, you may not see immediate need or good reason to, as the Options API works nicely.
+
+Our suggestion is to use this opportunity to think through the architecture and refactor any larger components that have too much responsibility on their own.
+
+For instance, if you have a component has the following methods:
+
+- one that gathers input from a form
+- one that validates that input
+- one that uses the input to call out to an API
+- one that does something with that API response
+
+You may find that you can potentially create a reusable component for the API call, as on a large application there's a fairly high chance you will need to make another.
+
+For the purposes of demonstration, here's how that one method refactor might look.
+
+### Before, with a Tightly Coupled Method
+
+```js
+methods: {
+  async getGitHubProjects() {
+    try {
+      this.isLoading = true
+      let projects = await fetch(
+        `https://api.github.com/users/sdras/repos?page=1&per_page=100`
+      ).then(res => res.json());
+
+      this.gitHubProjects = projects
+    } catch (error) {
+      this.error = error
+      console.error(error);
+    } finally {
+      this.isLoading = false;
+    }
+  }
+},
+created() {
+  this.getGitHubProjects()
+}
+```
+
+### After, with Composition API
+
+```js
+import { reactive, toRefs } from '@vue/composition-api'
```

Comment:
Vue 2 package.

Reply:
As mentioned above, the examples are Vue 2 intentionally because the aim is to show how to migrate an existing large Vue 2 application, not a greenfield Vue 3 application